compose/config/validation.py

```diff
@@ -216,15 +234,17 @@ def _parse_valid_types_from_schema(schema):
 
 def validate_against_fields_schema(config):
     schema_filename = "fields_schema.json"
-    return _validate_against_schema(config, schema_filename)
+    format_checkers = ["ports", "environment"]
+    return _validate_against_schema(config, schema_filename, format_checkers)
 
 
 def validate_against_service_schema(config, service_name):
     schema_filename = "service_schema.json"
-    return _validate_against_schema(config, schema_filename, service_name)
+    format_checkers = ["ports"]
```

Comment:
Thinking about this some more, I think there might be a better way to handle this. Instead of repeating the validation in the second schema, why not do something like this:

1) validate all the fields against `fields_schema.json`
2) remove the `$ref: fields_schema.json#/definitions/service` from `service_schema.json` and only validate the required fields exist. The other fields can be handled by `additionalProperties: true`, since we've already validated that all the fields are correct.

I think that way the format warning only logs once.

I'd actually be in favor of dropping support for abstract base services, but that's a topic for another time.